### ISSUE TITLE
[css-shadow-parts] Let ::theme() penetrate closed shadows

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#2381]
In <https://github.com/w3c/csswg-drafts/issues/2368#issuecomment-370168953>, Ryosuke comments:
> There should be no difference between closed and open shadow trees for ::theme. ::theme should always penetrate shadow boundaries. Otherwise, it defeats the primary use case of ::theme.

I would think that letting ::theme() penetrate closed shadows defeats the primary purpose of having a closed shadow in the first place.  I'm happy to go either way, tho.

**Comment by Jamesernator** [comment]
You have a note in [proposed spec](https://drafts.csswg.org/css-shadow-parts-1/)

> It’s important to note that ::part() and ::theme() offer absolutely zero new theoretical power.

However this isn't true, you already say in the following statement that you can avoid overstyling. This happens because `::part`/`::theme` *can't* accidentally inherit styles whereas custom properties can.

With that in mind, if `::theme` pierces shadow roots in the same way that custom variables do, is there any point in having `::theme` over just implementing `@apply`, as it seems to me that any `::theme(some-name)` would be identical to `@apply(--some-name)` under this shadow-piercing scheme.

**Comment by tabatkins** [comment]
No, @apply is still substantially different and suffers from several additional problems beyond what ::theme does. I outlined these in <https://www.xanthir.com/b4o00>.

**Comment by Jamesernator** [comment]
I see, `::theme` definitely seems like a better replacement for the functionality of `@apply`.

**Comment by equinusocio** [comment]
@Jamesernator  ...and @apply is officially deprecated and will never be supported :)
